drivers/max31855/max31855.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2024 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_max31855
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the drivers_sensors
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include "log.h"
+
+#include "max31855.h"
+#include "max31855_constants.h"
+#include "max31855_params.h"
+
+
+void _raw_data_to_thermocouple_temperature(uint32_t raw_data, int32_t *temperature)
+{
+    assert(temperature);
+
+    int32_t sign = (raw_data >> MAX31855_THERMOCOUPLE_SIGN_SHIFT) &
+                   MAX31855_THERMOCOUPLE_SIGN_MASK ? -1 : 1;
+
+    if (sign == -1) {
+        raw_data = ~raw_data + 1;
+    }
+
+    *temperature = (raw_data >>
+                    MAX31855_THERMOCOUPLE_INTEGER_SHIFT) & MAX31855_THERMOCOUPLE_INTEGER_MASK;
+
+    /* convert to centi degC */
+    *temperature = *temperature * 100;
+
+    /* add fractional parts */
+    *temperature += (raw_data & (1 << MAX31855_THERMOCOUPLE_FRACTIONAL_HALF_SHIFT)) ? 50 : 0;
+    *temperature += (raw_data & (1 << MAX31855_THERMOCOUPLE_FRACTIONAL_QUARTER_SHIFT)) ? 25 : 0;
+
+    *temperature *= sign;
+}
+
+void _raw_data_to_internal_temperature(uint32_t raw_data, int32_t *temperature)
+{
+    assert(temperature);
+
+    int32_t sign = (raw_data >> MAX31855_INTERNAL_SIGN_SHIFT) &
+                   MAX31855_INTERNAL_SIGN_MASK ? -1 : 1;
+
+    if (sign == -1) {
+        raw_data = ~raw_data + 1;
+    }
+
+    *temperature = (raw_data >> MAX31855_INTERNAL_INTEGER_SHIFT) & MAX31855_INTERNAL_INTEGER_MASK;
+
+    /* convert to centi degC */
+    *temperature = *temperature * 100;
+
+    /* add fractional parts */
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_HALF_SHIFT)) ? 50 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_QUARTER_SHIFT)) ? 25 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_EIGHTH_SHIFT)) ? 12 : 0;
+    *temperature += (raw_data & (1 << MAX31855_INTERNAL_FRACTIONAL_SIXTEENTH_SHIFT)) ? 6 : 0;
+
+    *temperature *= sign;
+}
+
+void _raw_data_to_fault(const uint32_t raw_data, max31855_fault_t *fault)
+{
+    assert(fault);
+
+    if (raw_data & (1 << MAX31855_FAULT_VCC_SHORT_SHIFT)) {
+        *fault = MAX31855_FAULT_VCC_SHORT;
+    }
+    else if (raw_data & (1 << MAX31855_FAULT_GND_SHORT_SHIFT)) {
+        *fault = MAX31855_FAULT_GND_SHORT;
+    }
+    else if (raw_data & (1 << MAX31855_FAULT_OPEN_CIRCUIT_SHIFT)) {
+        *fault = MAX31855_FAULT_OPEN_CIRCUIT;
+    }
+    else {
+        *fault = MAX31855_FAULT_NO_FAULT;
+    }
+}
+
+int max31855_init(max31855_t *dev, const max31855_params_t *params)
+{
+    assert(dev && params);
```

Comment:
```suggestion
    assert(dev);
    assert(params);
```

Ensure that different failure modes result in different assert failure messages / addresses.